lib/src/json_schema/json_schema.dart

```diff
@@ -631,7 +631,17 @@ class JsonSchema {
           }
 
           // If currentSchema is a ref, we might want to resolve it recursively right now.
-          if (currentSchema.ref != null && currentSchema._schemaMap.length == 1) {
+          if (currentSchema.ref != null) {
+            // If we are at the end of the fragments to search and there are additional properties in the schema,
+            // continue with the current schema instead of resolving the ref.
+            if (i + 1 == fragments.length && currentSchema._schemaMap.length > 1) {
+              continue;
+            }
+            // If the next fragment is in the current schema, continue with the current schema instead of resolving the ref.
+            if (i + 1 < fragments.length && currentSchema._schemaMap.containsKey(fragments[i + 1])) {
+              continue;
+            }
```

Comment:
It's really nice to have this spell out clearly, thanks! You could inline the conditions and comments into the `if`, but this works too.

Reply:
I initially inlined them, but I think breaking the `if` statement up and adding comments helps readability a bit.